craterpy/classes.py

```diff
@@ -469,3 +475,186 @@ def plot(self, name="", ax=None, alpha=0.2, **kwargs):
+        
+        # Convert to the target CRS if specified
+        if crs is not None:
+            # Handle various input types for CRS
+            try:
+                target_crs = CRS.from_user_input(crs)
+                if data.crs != target_crs:
+                    data = data.to_crs(target_crs)
+            except Exception as e:
+                raise ValueError(f"Error converting to CRS '{crs}': {str(e)}")
+        
+        # Export to GeoJSON
+        if filename is not None:
+            data.to_file(filename, driver="GeoJSON")
+            return None
+        else:
+            # Return as string
+            return data.to_json()
+
+
+    @classmethod
+    def read_shapefile(cls, filename, body="Moon", units="m"):
+        """
+        Read crater data from a shapefile or GeoJSON file.
+        
+        Parameters
+        ----------
+        filename : str
+            Path to the shapefile or GeoJSON file.
+        body : str, optional
+            Planetary body, e.g. Moon, Vesta (default: Moon).
+            If the file contains a 'body' or 'planet' field, that value will be used instead.
+        units : str, optional
+            Length units of radius/diameter, m or km (default: m).
+            If the file contains a 'units' field, that value will be used instead.
+        
+        Returns
+        -------
+        CraterDatabase
+            A new CraterDatabase instance containing the data from the file.
+            
+        Notes
+        -----
+        This method assumes the file was previously created by CraterDatabase.to_geojson()
+        or has a compatible format with lat/lon coordinates and radius or diameter information.
+        
+        If the file contains different coordinate column names than expected, this method
+        will attempt to identify them by common names (e.g., 'lat', 'latitude', 'lon', 'longitude').
+        """
+        gdf = gpd.read_file(filename)
+        
+        # Check if the file has body or units info
+        if 'body' in gdf.columns and gdf['body'].nunique() == 1:
+            body = gdf['body'].iloc[0]
+        elif 'planet' in gdf.columns and gdf['planet'].nunique() == 1:
+            body = gdf['planet'].iloc[0]
+            
+        if 'units' in gdf.columns and gdf['units'].nunique() == 1:
+            units = gdf['units'].iloc[0]
+        
+        # Create a working copy to modify
+        data = gdf.copy()
+        
+        # Try to identify latitude column
+        lat_col = None
+        lat_candidates = ['lat', 'latitude', 'latitude_deg', 'lat_deg', 'y']
+        for col in lat_candidates:
+            if col in data.columns:
+                lat_col = col
+                break
+        
+        # Try to identify longitude column
+        lon_col = None
+        lon_candidates = ['lon', 'longitude', 'longitude_deg', 'lon_deg', 'x']
+        for col in lon_candidates:
+            if col in data.columns:
+                lon_col = col
+                break
```

Comment:
use `craterpy.helper.get_crater_cols()`. don't think we need to support  'x' and 'y' until there is a need, and the rest should be handled